### PR TITLE
chore: replace deprecated function for debugging

### PR DESCRIPTION
**Pull request by cristiGuranIonos**
## Checklist

- [ ] PR name added as appropriate (e.g. `feat:`/`fix:`/`doc:`/`test:`/`refactor:`)
- [ ] Tests added or updated
- [ ] Documentation updated
- [ ] Changelog updated and version incremented (label: upcoming release)
- [ ] Github Issue linked if any
- [ ] Jira task updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 - Add `nic` service
 - Use `error.As` for `requestFailed` err
 - `%w` instead of `%s` for some printed errors
+- use `serve` for debug mode
```
Comment (adeatcu-ionos) [inline CHANGELOG.md:6]: @cristiGuranIonos not important at all but we should be consistent about whether we use capital letter for the first word of the list or not, maybe we can do that in another PR.

File: main.go [hunk 2]
```diff
@@ -18,17 +17,12 @@ func main() {
...
-			})
-		if err != nil {
-			log.Println(err.Error())
-		}
-	} else {
-		plugin.Serve(&plugin.ServeOpts{
-			ProviderFunc: ionoscloud.Provider,
-		})
+		serveOpts.ProviderAddr = "registry.terraform.io/ionos-cloud/ionoscloud"
+		serveOpts.Debug = debugMode
```
Comment (adeatcu-ionos) [inline main.go:25]: @cristiGuranIonos again, not something important but for a little bit of clarity you can do:
`serveOpts.Debug = true` but it's not a big deal, do as you feel.

Comment (cristiGuranIonos) [reply main.go:25]: will leave it as it is for now

File: CHANGELOG.md (hunk 1 above)
Comment (cristiGuranIonos) [reply CHANGELOG.md]: done

**Review by adeatcu-ionos — APPROVED** [review]
